crates/flux-refineck/src/checker.rs

```diff
@@ -845,6 +845,13 @@ impl<'ck, 'genv, 'tcx, M: Mode> Checker<'ck, 'genv, 'tcx, M> {
                         idx.clone()
                     }
                 }
+                TyKind::Indexed(BaseTy::Char, idx) => {
+                    if bits == 0 {
+                        idx.not()
+                    } else {
+                        idx.clone()
+                    }
```

Comment:
I cannot figure out what `bits` means.

Reply:
@Samir-Rashid a `switchInt(_1)` roughly means something like

```rust
if _1 == v0 {
   goto bb0
} else if _1 == v1 {
   goto bb1
} else if _1 == v2 {
   ...
} else {
   goto bbn
}
```
i.e., test if `_1` equals a specific value and jump to a new basic block. The `bits` are the bits that make up those values. The content of the bits and what they mean depends on the type of `_1` (e.g., for `bool` the bits can be either 0 or 1)

When we check `switchInt` we try to gain control flow information from assuming the test succeeded. For example, assuming `_1` has type `i32[a]` and we giving the sequence
```rust
if _1 == 0 {
    goto bb0
} else {
   goto bb1
 }
```
we want to check `bb0` (resp. `bb1`) knowing that `a == 0` (resp. `a != 0`). This is captured by associating each successor basic block with `Guard::Pred(p)` where `p` represents the predicate we can assume when checking the basic block.

`char`'s are currently unrefined so there's no control flow information we can assume when checking successors (all successors should be associated with `Guard::None`). So for `chars` we have to ignore the bits.